src/reducers/event.js

```diff
@@ -1,31 +1,45 @@
 export const SET_EVENTS = "SET_EVENTS";
+export const GET_EVENTS_ERROR = "GET_EVENTS_ERROR";
 import eventServices from "services/event";
 
-const initialState = [];
+const initialState = {
+  events: [],
+  error: null
+};
 
 export default function eventReducer(state = initialState, action) {
   switch (action.type) {
     case SET_EVENTS:
-      return action.payload;
+      return { ...state, error: null, events: action.payload };
+    case GET_EVENTS_ERROR:
+      return { ...state, error: action.payload.error };
```

Comment:
Would it be conventional to always set some value for `events` so that they keys are the same for every case in the switch statement? Even if that value is `null`? I also like when the lack of a collection is modeled as an empty array, because it makes it more likely that ui depending on it won't break.